app/src/main/java/com/jetbrains/handson/mpp/mobile/AdvancedSearchActivity.kt

```diff
@@ -93,15 +94,16 @@ class AdvancedSearchActivity : AppCompatActivity(), AdvancedSearchContract.View,
         val intent = Intent(this, MainActivity::class.java)
         intent.putExtra("numAdults", getNumAdults() )
         intent.putExtra("numChildren", getNumChildren())
-        if (timeChanged){
-            intent.putExtra("time", getDateInAPIFormat())
-        } else {
-            intent.putExtra("time", "Unset")
-        }
+        intent.putExtra("time", getDateInAPIFormat())
+        intent.putExtra("timeText", getTimeText())
         setResult(2, intent)
         finish()
     }
     private fun getDateInAPIFormat(): String {
```

Comment:
There's some commonality between this method and the `getTimeText` method, maybe you could have a method called something like `getTime` that returns a DateTime object, which both of these functions can use, and then can use different DateTimeFormats to create the strings you need?